web/wp-content/plugins/tmbi-adobe-dtm/inc/adobe_dtm_utils.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/*
+ * User: jeysaravana
+ * Date: 2018-01-29
+ * Time: 12:18 PM
+ */
+
+class Adobe_DTM_Utils {
+	const PUBLISH_META_NAME = 'original_publish_date';
+
+	public $registered_post_type = array(
```

Comment:
This knowledge about the environment works against maintainability, there's more than one reason to change this class (the functionality to store the meta changes OR a new post type is added).

An option to isolate/decouple this code from the environment is are [add_post_type_support](https://developer.wordpress.org/reference/functions/add_post_type_support/) and [get_post_types_by_support](https://developer.wordpress.org/reference/functions/get_post_types_by_support/) functions provided by WordPress. This class can provide a functionality ("store original published date as meta") to any of the registered post types that support (or want to use) it.

Reply:
See https://readersdigest.atlassian.net/browse/DRDT-101